### ISSUE TITLE
FAIR validation results seem wrong

### ISSUE THREAD

**Issue by yafeim** [HavenFeng/TRUST#8]
Hi,

Thanks for providing the code. I tested your model on the provided FAIR val dataset. I use 'python test.py --test_folder TRUST_models_BFM_version --test_split val' command to run the code. It does show some warnings/print msgs (pasted below), the code can be run with no problems. And I got the following results:

<img width="865" alt="image" src="https://user-images.githubusercontent.com/29441946/214914146-3db747c2-aad2-4305-9dff-44cb8db0bf3a.png">. 

It seems that the model has a strong bias towards lighter skin tone. Can you advise? Thanks. 

fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
copy resnet state dict finished!
fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
copy resnet state dict finished!
fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
copy resnet state dict finished!
fc.weight  not available in reconstructed resnet
fc.bias  not available in reconstructed resnet
copy resnet state dict finished!
creating the FLAME Decoder
/mnt/TRUST/lib/nets/FLAME.py:89: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  self.register_buffer('dynamic_lmk_faces_idx', torch.tensor(lmk_embeddings['dynamic_lmk_faces_idx'], dtype=torch.long))
/mnt/TRUST/lib/nets/FLAME.py:90: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  self.register_buffer('dynamic_lmk_bary_coords', torch.tensor(lmk_embeddings['dynamic_lmk_bary_coords'], dtype=self.dtype))
trained model found. load ./data/deca_model.tar
trained model found. load TRUST_models_BFM_version/E_scene_light_BFM.tar
trained model found. load TRUST_models_BFM_version/E_face_light_BFM.tar
trained model found. load TRUST_models_BFM_version/E_albedo_BFM.tar
/mnt/env/trust_py38/lib/python3.8/site-packages/pytorch3d/io/obj_io.py:546: UserWarning: Mtl file does not exist: ./data/template.mtl
  warnings.warn(f"Mtl file does not exist: {f}")

**Comment by HavenFeng** [comment]
Hey, thanks for pointing out this issue. I've looked into it and found it's partially due to the wrong cropping window of the scene image (the dataloader code will be updated soon), and **mainly due to the limitations of BFM albedo space (Please be aware that this is not our main model).** This is like the furthest BFM-based regressor can reach for representing dark skin tone.

It also aligns with what we've found and written into the paper, that the BFM version has a very high error for type 6 skin (where most dark skin tone samples are located). Consequently, there's no ambiguity to regularize since the albedo space is heavily biased. The light/albedo ambiguity only happens when the albedo space is diverse enough.  Below you can see the comparison of this BFM albedo based model (left) and our main model (right, trained jointly with our diverse albedo space)
![comparison_image](https://user-images.githubusercontent.com/16853754/216959997-eb5fe788-569c-42b4-aab3-6907511e8f85.png)

You can see that not only our albedo estimation is more accurate, the light prediction is a lot better.
We are working closely with the data vendor (who provided us the albedo space data) on releasing the diverse albedo space and our paper main model (which trained with the diverse space). Stay tuned~
